reflex/components/component.py

```diff
@@ -271,11 +277,18 @@ def __init__(self, *args, **kwargs):
                 field_type = EventChain
             elif key in props:
                 # Set the field type.
-                field_type = fields[key].type_
+                field_type = fields[key].annotation
```

Comment:
in other places in the code `annotation` seems to be a function? 🤔 not sure if this is incorrect, but leaving myself a bookmark with this comment

Reply:
Yeah, `cs_obj = field.annotation()` is there to initialize default values iirc. Maybe pydantic provides smth like `get_field_default` which handles `default` and `default_factorys` for us?